server/protocol/qdatastream.py

```diff
@@ -95,7 +95,11 @@ async def read_message(self):
         if action in ("PING", "PONG"):
             return {"command": action.lower()}
 
-        message = json.loads(action)
+        try:
+            message = json.loads(action)
+        except json.decoder.JSONDecodeError as err:
+            raise json.decoder.JSONDecodeError(
+                "Invalid JSON, full action string was: {}".format(action)) from err
```

Comment:
Use f-string please, and I think we use `e` as the error name in other places so it would be good to do that here too.

Reply:
done and done